### ISSUE TITLE
Login/register form missing bottom border on Twenty Seventeen theme

### ISSUE THREAD

**Issue by sumobi** [AffiliateWP/AffiliateWP#1919]
Some form fields are missing a bottom border on the Twenty Seventeen theme:

![screen shot 2017-01-13 at 2 29 20 pm](https://cloud.githubusercontent.com/assets/52581/21915149/c71ea37c-d99c-11e6-8d62-e1bc9f462152.png)

This is because the theme has this CSS rule:

```
.required[title] {
    border-bottom: none;
}
```

**Comment by sumobi** [comment]
There's no way to override that rule via CSS so the bottom border appears again without defining a border color. This is not something we would do.

My proposed solution would be to remove the `required` CSS class from each of our fields. They're not used by validation and since it's now possible for any form to essentially be required or not required (as of new v2.0 checkbox options) there's no point in keeping them around.

@pippinsplugins Do you agree? Any purpose of those CSS classes I'm missing?

It also means we can then remove these lines of CSS:
https://github.com/AffiliateWP/AffiliateWP/blob/master/assets/css/forms.css#L53-L58

**Comment by pippinsplugins** [comment]
@sumobi If we remove the class then we should add in the `required` attribute for HTML5 validation.

**Comment by sumobi** [comment]
@pippinsplugins Good point. We'll also need to dynamically insert the required attribute based on the checkboxes selected.

**Comment by pippinsplugins** [comment]
Punting this to 2.0.1 then.

**Comment by DrewAPicture** [comment]
@sumobi Have you followed up on the HTML5 validation in the PR yet? Just triaging 2.0.1.

**Comment by sumobi** [comment]
@DrewAPicture I have not.

**Comment by DrewAPicture** [comment]
Punting to 2.0.3 pending a requested update for HTML5 validation.

**Comment by DrewAPicture** [comment]
@sumobi Mind following up here so we can get this closed out?

**Comment by sumobi** [comment]
PR: https://github.com/AffiliateWP/AffiliateWP/pull/2063 (CSS will need to be minified at release)

@pippinsplugins @DrewAPicture Turns out we can't use the `required` attribute since we use `novalidate` on the `form` tag. The HTML5 validation will not work.

```
<form id="affwp-register-form" class="affwp-form" action="" method="post" novalidate="novalidate">
```

This was due to https://github.com/AffiliateWP/AffiliateWP/issues/1792

Not sure if there's a workaround to have both `required` attributes and proper validation on the URL field. I vote we just take the required classes out for now to fix this issue and then sit on it for a while. We've still got validation, just not HTML5 validation and I'd rather not have the URL field not validating correctly.

**Comment by pippinsplugins** [comment]
>>>I vote we just take the required classes out for now to fix this issue and then sit on it for a while.

I'm fine with this.

**Comment by DrewAPicture** [comment]
New PR: #2243 

It implements HTML5 validation on the registration form. Introduces a new `affwp_required_field_attr()` function + tests that handles outputting the `required='required'` attribute if the registration field is in fact required in settings.

**Comment by sumobi** [comment]
@DrewAPicture Nice!

Lets add the `required` attribute to the terms checkbox since that will always be required if it is shown. Currently, if you fill in everything else but forget to check the terms box, it will submit the form and you lose what you typed into the password fields.

**Comment by DrewAPicture** [comment]
Added the `required` attribute to the TOS checkbox. Waiting for tests to pass then I'll merge it.

**Comment by sumobi** [comment]
👍

**Comment by DrewAPicture** [comment]
Merged #2243.
